std/uni.d

```diff
@@ -7787,6 +7787,41 @@ if (isInputRange!S1 && isSomeChar!(ElementEncodingType!S1)
     import std.typecons : Yes;
     static import std.ascii;
 
+    static if ((isDynamicArray!S1 || isRandomAccessRange!S1)
+        && (isDynamicArray!S2 || isRandomAccessRange!S2)
+        && !(isInfinite!S1 && isInfinite!S2)
+        && __traits(compiles,
+            {
+                size_t s = size_t.sizeof / 2;
+                r1 = r1[s .. $];
+                r2 = r2[s .. $];
+            }))
+    {{
+        // ASCII optimization for dynamic arrays & similar.
+        size_t i = 0;
+        static if (isInfinite!S1)
+            immutable end = r2.length;
+        else static if (isInfinite!S2)
+            immutable end = r1.length;
+        else
+            immutable end = r1.length > r2.length ? r2.length : r1.length;
+        for (; i < end; ++i)
+        {
+            auto lhs = r1[i];
+            auto rhs = r2[i];
+            if ((lhs | rhs) >= 0x80) goto non_ascii_path;
```

Comment:
Nit: I couldn't many precedents for using `_` in labels (there are a few `L_backtrack`).
IIRC Andrei requested once that labels should start with UpperCase too, so it would be `NonAsciiPath`, but I'm not entirely sure on this either. We should probably put it to the DStyle once clarified.

Reply:
Fixed.